plugins/perceptionexpander.koplugin/main.lua

```diff
@@ -0,0 +1,232 @@
+                    self.line_thick),
+            },
+            {
+                text = "",
+                input_type = "number",
+                hint = T(_("Margin from edges. Current value: %1"),
+                    self.margin),
+            },
+            {
+                text = "",
+                input_type = "number",
+                hint = T(_("Line color intensity (1-10). Current value: %1"),
+                    self.line_color_intensity * 10),
+            },
+            {
+                text = "",
+                input_type = "number",
+                hint = T(_("Increase margin after pages. Current value: %1"),
+                    self.shift_each_pages),
+            },
+        },
+        buttons ={
+            {
+                {
+                    text = _("Cancel"),
+                    callback = function()
+                        self.settings_dialog:onClose()
+                        UIManager:close(self.settings_dialog)
+                    end
+                },
+                {
+                    text = _("Apply"),
+                    callback = function()
+                        self:saveSettings(MultiInputDialog:getFields())
+                        self.settings_dialog:onClose()
+                        UIManager:close(self.settings_dialog)
+                        self:createUI()
+                    end
+                },
+            },
+        },
+        width = Screen:getWidth() * 0.8,
+        height = Screen:getHeight() * 0.3,
+    }
+    self.settings_dialog:onShowKeyboard()
+    UIManager:show(self.settings_dialog)
+end
+
+function PerceptionExpander:addToMainMenu(tab_item_table)
+    table.insert(tab_item_table.plugins,{
+        text = _("Speed reading module - perception expander"),
+        sub_item_table ={
+            {
+                text = _("Enable"),
+                checked_func = function() return self.is_enabled end,
+                callback = function()
+                    self.is_enabled = not self.is_enabled
+                    self:saveSettings()
+                    return true
+                end,
+            },
+            {
+                text = _("Settings"),
+                callback = function()
+                    self:showSettingsDialog()
+                end,
+            },
+            {
+                text = _("About"),
+                callback = function()
+                    UIManager:show(InfoMessage:new{
+                        text = _("For more information see wiki page Perception Expander Plugin"),
+                    })
+                end,
+            },
+        },
+    })
+end
+
+-- in case when screensaver starts
+function PerceptionExpander:onSaveSettings()
+    self:saveSettings()
+end
+
+function PerceptionExpander:onPageUpdate(pageno)
+    if not self.is_enabled then
+        return
+    end
+
+    if self.page_counter >= self.shift_each_pages and self.margin < 0.37 then
```

Comment:
It's a better practice to define magic numbers as module global variables with descriptive names.